tests/unit/multiclass/KNN_unittest.cc

```diff
@@ -8,98 +8,118 @@
 
 using namespace shogun;
 
-#ifdef HAVE_LAPACK
-void generate_knn_data(SGMatrix<float64_t>& feat, SGVector<float64_t>& lab,
-	   	int32_t num, int32_t classes, int32_t feats)
+class KNNTestData
 {
-	CMath::init_random(1);
-	feat = CDataGenerator::generate_gaussians(num,classes,feats);
-	for( int i = 0 ; i < classes ; ++i )
-		for( int j = 0 ; j < num ; ++j )
-			lab[i*num+j] = double(i);
+	public:
+		KNNTestData() 
```

Comment:
adding unit tests should be different PRs than fixing memory leaks
(unless the test you added covers a bug you fixed)